compilation: remove_if

@bhatiaharsh and I ran into this whilst running `spack install flux-sched@master` on TOSS and RHEL systems.  I have a commit in #385 that fixes this problem, but in the interest of getting this fix merged ASAP, I'll spin it out into a separate PR.

```
/home/herbein1/Repositories/spack/var/spack/stage/flux-sched-master-jiemnn7kiqbnegkl6qeifknungdys4tx/flux-sched/resource/policies/base/matcher.cpp: In member function 'int Flux::resource_model::matcher_util_api_t::register_resource_pair(const string&, std::__cxx11::string&)':
/home/herbein1/Repositories/spack/var/spack/stage/flux-sched-master-jiemnn7kiqbnegkl6qeifknungdys4tx/flux-sched/resource/policies/base/matcher.cpp:167:14: error: 'remove_if' is not a member of 'std'
     h.erase (std::remove_if (h.begin (), h.end (), ::isspace), h.end ());
              ^~~
/home/herbein1/Repositories/spack/var/spack/stage/flux-sched-master-jiemnn7kiqbnegkl6qeifknungdys4tx/flux-sched/resource/policies/base/matcher.cpp:175:14: error: 'remove_if' is not a member of 'std'
     l.erase (std::remove_if (l.begin (), l.end (), ::isspace), l.end ());
              ^~~
```

--- a/resource/Makefile.am
+++ b/resource/Makefile.am
@@ -6,8 +6,7 @@ AM_CXXFLAGS = \
     -Wno-error \
     $(CODE_COVERAGE_CXXFLAGS)
 
-AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS) \
-             $(BOOST_LDFLAGS)
+AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS)
 
 AM_CPPFLAGS = -I$(top_srcdir) $(CZMQ_CFLAGS) $(FLUX_CORE_CFLAGS) \
 	      $(BOOST_CPPFLAGS)
@@ -69,6 +68,7 @@ libresource_la_CXXFLAGS = \
 libresource_la_LIBADD = \
     $(top_builddir)/resource/planner/libplanner.la \
     $(top_builddir)/resource/libjobspec/libjobspec_conv.la \
+    $(BOOST_LDFLAGS) \
     $(BOOST_SYSTEM_LIB) \
     $(BOOST_FILESYSTEM_LIB) \
     $(BOOST_GRAPH_LIB) \

--- a/resource/modules/Makefile.am
+++ b/resource/modules/Makefile.am
@@ -6,8 +6,7 @@ AM_CXXFLAGS = \
     -Wno-error \
     $(CODE_COVERAGE_CXXFLAGS)
 
-AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS) \
-             $(BOOST_LDFLAGS)
+AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS)
 
 AM_CPPFLAGS = -I$(top_srcdir) $(CZMQ_CFLAGS) $(FLUX_CORE_CFLAGS) \
           $(BOOST_CPPFLAGS)
@@ -30,7 +29,8 @@ resource_la_LIBADD = \
     $(HWLOC_LIBS) \
     $(UUID_LIBS) \
     $(JANSSON_LIBS) \
-    $(CZMQ_LIBS)
+    $(CZMQ_LIBS) \
+    $(BOOST_LDFLAGS)
 resource_la_LDFLAGS = \
     $(AM_LDFLAGS) \
     $(fluxmod_ldflags) -module

--- a/resource/policies/base/matcher.cpp
+++ b/resource/policies/base/matcher.cpp
@@ -22,6 +22,8 @@
  *  See also:  http://www.gnu.org/licenses/
 \*****************************************************************************/
 
+#include <algorithm>
+
 #include "policies/base/matcher.hpp"
 
 namespace Flux {

--- a/resource/utilities/Makefile.am
+++ b/resource/utilities/Makefile.am
@@ -6,8 +6,7 @@ AM_CXXFLAGS = \
     -Wno-error \
     $(CODE_COVERAGE_CXXFLAGS)
 
-AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS) \
-             $(BOOST_LDFLAGS)
+AM_LDFLAGS = $(CODE_COVERAGE_LDFLAGS)
 
 AM_CPPFLAGS = -I$(top_srcdir) $(CZMQ_CFLAGS) $(FLUX_CORE_CFLAGS) \
           $(BOOST_CPPFLAGS)
@@ -20,6 +19,7 @@ grug2dot_CXXFLAGS = \
     $(AM_CXXFLAGS)
 grug2dot_LDADD = \
     ../libresource.la \
+    $(BOOST_LDFLAGS) \
     $(BOOST_SYSTEM_LIB) \
     $(BOOST_FILESYSTEM_LIB) \
     $(BOOST_GRAPH_LIB) \
@@ -37,4 +37,3 @@ resource_query_LDADD = \
     $(BOOST_FILESYSTEM_LIB) \
     $(BOOST_GRAPH_LIB) \
     $(BOOST_REGEX_LIB)
-

